plantswap/components/Uploader/Uploader.js

```diff
@@ -9,118 +9,144 @@ import Title from "../Title/Title";
-  const [loading, setLoading] = useState(false);
+  const { flowData, setFlowData } = flowdata;
+  const [uploadedValue, setUploadedValue] = useState(
+    flowData.plant.url ? 100 : 0
+  );
+  const [data, setData] = useState(dummydata);
   const inputFileRef = useRef(null);
   const API_URL = "/api/identify";
-  console.log("render");
 
-  const fetchData = (e, abort) => {
-    if (e?.target?.files && e?.target?.files[0] && !abort) {
+  console.log(uploadedValue);
+  const fetchData = (e) => {
+    if (e?.target?.files && e?.target?.files[0]) {
+      var formdata = new FormData();
       let xml = new XMLHttpRequest();
+      let filereader = new FileReader();
       xml.open("POST", API_URL, true);
       const file = e.target.files[0];
-      var formdata = new FormData();
       formdata.append("image", file);
       formdata.append("lang", locale);
       xml.upload.addEventListener("progress", progressHandler, false);
 
-      setCurrentFetch(xml);
-      setImage(file);
-      setLoading(true);
+      filereader.addEventListener("load", (event) => {
+        const result = event.target.result;
+        // Do something with result
+        const newFile = {
+          url: result,
+          name: file.name,
+          type: file.type,
+          size: file.size,
+        };
 
-      xml.send(formdata);
-      xml.onreadystatechange = function (e) {
-        function isJSON(jsonString) {
-          try {
-            var o = JSON.parse(jsonString);
-            if (o && typeof o === "object") {
-              return o;
-            }
-          } catch (e) {
-            console.log("not valid json");
-          }
+        setFlowData((prev) => {
+          return { ...prev, plant: newFile };
+        });
+
+        setCurrentFetch((prev) => {
+          console.log(prev);
+          return xml;
+        });
+      });
 
-          return false;
+      filereader.addEventListener("progress", (event) => {
+        if (event.loaded && event.total) {
+          const percent = (event.loaded / event.total) * 100;
+          console.log(`Progress: ${Math.round(percent)}`);
+          //     setUploadedValue(Math.round(percent));
         }
+      });
 
+      filereader.readAsDataURL(file);
+      xml.send(formdata);
+
+      xml.onreadystatechange = function (e) {
         if (isJSON(e.target.response) && e.target.readyState === 4) {
-          console.log(isJSON(e.target.response));
           const { data } = isJSON(e.target.response);
-          setData(data);
+          setFlowData((prev) => {
+            return { ...prev, apidata: data };
+          });
         }
       };
-    } else if (abort && currentFetch) {
-      currentFetch.abort();
-
-      if (image && uploadedValue === 100) {
-        inputFileRef.current.value = "";
-        e.preventDefault();
-        setImage(undefined);
-        setUploadedValue(0);
-        setData(undefined);
-        setShowData(false);
-      }
     }
   };
 
+  console.log(currentFetch);
```

Comment:
Spotted a console.log here. It's usually best to remove these before production for cleaner, more efficient code.